### DISCUSSION TITLE
Simulation error: Solver failed because the following interpolation bounds were exceeded

### DISCUSSION THREAD

**Discussion by mlnsai** [pybamm-team/PyBaMM#2714] [Q&A]
Hi Team,

I am performing drive cycle simulation for one year to understand the degradation phenomenon. I have enabled Mechanical as well as Chemical degradation packages viz. SEI layer growth, Plating and Particle Cracking.

I have tried the simulation with three drive cycles; (UDDS and Two custom drive cycles)

Every time I am getting the same following error; 

**Simulation error: Solver failed because the following interpolation bounds were exceeded: ["Interpolant 'interpolating_function' upper bound"]. You may need to provide additional interpolation points outside these bounds.**

UDDS; Error after 18/365 cycles
Drive cycle;1 : Error after 18/365 cycles
Drive cycle;1 : Error after 68/365 cycles

I couldn't figure out which bound I am exceeding. Please have a look :)

@tinosulzer @brosaplanella 

Thanks in advance :)

Posting my code:

########## Model and Parameter slection ############
```
model=pybamm.lithium_ion.SPMe(options=option)
param1 = pybamm.ParameterValues("OKane2022")
```
########## Options ##############
```
option={
"SEI":"solvent-diffusion limited",
"particle mechanics": ("swelling and cracking", "swelling only"),
"particle" : "Fickian diffusion",
"SEI on cracks": "true",
"loss of active material": "stress and reaction-driven",
"thermal": "lumped",
"SEI film resistance": "distributed",
"SEI porosity change" : "true", 
"lithium plating": "reversible",
}
```
############ Experiment ###########
```
cycle = 365
Exp = [(
                "Run Drive_Cycle (A)", 
                "Rest for 9 hours", 
                "Run Drive_Cycle (A)",
                "Rest for 5 minutes",
                "Charge at 0.1C until 4.2V",
                "Hold at 4.2V for 2 hours",
                "Rest for 2 hours"
            )]* cycle
```
################# Solver #########################
```
solver=pybamm.CasadiSolver("safe",dt_max=200,max_step_decrease_count=20)
var_pts={"x_n": 40, "x_s":20, "x_p": 40, "r_n": 40, "r_p":40}
sim=pybamm.Simulation(model, experiment = experiment,parameter_values = param1,solver = solver,var_pts=var_pts)
solution=sim.solve(calc_esoh=False)
```

**Comment by cannotcomplain** [comment]
I had this error recently and it was because the OCP interpolant I had created did not extend all the way to 0 and 1 stoichiometry (it was close, about 1e-3 and 0.999, but not 0 or 1). Adding points at 0 and 1 fixed it. I am not sure how the O'Kane 2022 OCPs are defined but this might be the issue.

**Reply by mlnsai** [reply]
Hi @cannotcomplain ,

Thanks for the reply. After your suggestion I did checked the OCP data for O'Kane 2022 set.

Only for graphite, they are referring to this csv file "[graphite_LGM50_ocp_Chen2020.csv](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/data/graphite_LGM50_ocp_Chen2020.csv)", where Interpolation is being done.

```
path, _ = os.path.split(os.path.abspath(__file__))
graphite_LGM50_ocp_Chen2020_data = pybamm.parameters.process_1D_data(
    "graphite_LGM50_ocp_Chen2020.csv", path=path
)


def graphite_LGM50_ocp_Chen2020(sto):
    name, (x, y) = graphite_LGM50_ocp_Chen2020_data
    return pybamm.Interpolant(x, y, sto, name=name, interpolator="cubic")
```

Recently, they did add more points to O'Kane 2022 OCP data, where the extreme points are covered (0 and 1). and for NMC, they are using an empirical relation for ocp and sto; So I guess that's not an issue.

```
    u_eq = (
        -0.8090 * sto
        + 4.4875
        - 0.0428 * pybamm.tanh(18.5138 * (sto - 0.5542))
        - 17.7326 * pybamm.tanh(15.7890 * (sto - 0.3117))
        + 17.5842 * pybamm.tanh(15.9308 * (sto - 0.3120))
    )
    return u_eq
```

I did add few more points for graphite csv file in between and changed the interpolation to cubic spline in code and checked. Still I am getting the same error. :/

Should I look elsewhere? 

Thanks for your time :)

**Reply by manjunathnilugal** [reply]
Hi, 
@mlnsai Even i am facing similar issue with after certain set of drive cycle simulation. Is there any method to resolve this issue.

**Reply by brosaplanella** [reply]
There could be many causes to this error, please share a minimum working example and the full error log.
